### ISSUE TITLE
Spurious parse error when destructuring an assignment

### ISSUE THREAD

**Issue by josephg** [intellij-rust/intellij-rust#9670]
## Environment

* **IntelliJ Rust plugin version:** 0.4.181.4965-222
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.2.3 (CL-222.4167.35)
* **Operating system:** Linux 5.15.0-52-generic
* **Macro expansion:** enabled

## Problem description

The code below shows error squiggly lines in the IDE but compiles in rustc just fine.

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8994d09798723f01eaf7f206e915fa1d)

## Steps to reproduce

Code:

```rust
fn foo() {
    struct Container(Option<(String, String)>);

    let mut entry = Container(Some(("hi".into(), "yo".into())));
    if let Some(e) = entry.0.as_mut() {
        (e.0, e.1) = ("xxx".into(), "yyy".into()); // Errors "Cannot move"

        (e.0, _) = ("xxx".into(), 123); // Also errors (independently): "_ or ) expected"
    }
}
```

![image](https://user-images.githubusercontent.com/47413/199630636-68dfc13c-1c10-468a-916d-7b897bc78e59.png)
